docs/architecture-decision-records/005-x01-package-structure.md

```diff
@@ -0,0 +1,102 @@
+- `cdk-experiments` for experimental code (published)
+
+#### Score
+- :white_check_mark: Contributing new experimental features
+
+  Use of Workspaces and shared code would provide a familiar DX to the current repository, in that typechecking, linting, testing etc. will continue to be available.
+
+- :white_check_mark: Consuming an experimental feature
+
+  Consumption would require explicit installation of a new NPM package, making it clear that an experimental feature is in use:
+
+  ```ts
+  import { StableFeature } from "@guardian/cdk";
+  import { EarlyFeature } from "@guardian/cdk-experiments";
+   ```
+
+  There may be compatability complexities in this solution.
+  Today, when a project uses `@guardian/cdk`, the version of AWS CDK in use must _exactly match_ those used by `@guardian/cdk`.
+  If `@guardian/cdk-experiments` uses a later version of AWS CDK, consuming projects must first update `@guardian/cdk` before they can make use of `@guardian/cdk-experiments`.
+  Use of `peerDependencies` would likely solve this, however.
+
+- :x: Long term maintenance
+
+  NPM Workspaces can yield complex directory structures and configuration, in particular when creating a workspace for shared code.
+
+  The release process could get complex too, in particular around automation.
+  Today, releases are automated using [semantic-release](https://github.com/semantic-release/semantic-release), this handles the creation of structured release notes based on commit messages.
+  In a world of multiple, packages, should there be one set of release notes that details all the changes? Or a set of release notes per package? And does semantic-release support this?
+
+  There are ambitions to adopt [JSII](https://github.com/aws/jsii) for cross language publishing. Would we publish both packages? Or just the core package?
+
+### 2. Host experiments in an `experimental` directory
+Published within the `@guardian/cdk` package, experimental features will be defined within an `experimental` directory.
+
+#### Score
+- :white_check_mark: Contributing new experimental features
+
+  Contributing new experimental features will be similar to contributing stable features, with the only difference being the directory.
+  Typechecking, linting, testing etc. will continue to be available.
+
+- :white_check_mark: Consuming an experimental feature
+
+  As experimental features are published directly within the `@guardian/cdk` package, consumption would look similar to any other pattern or construct:
+
+  ```ts
+  import { StableFeature } from "@guardian/cdk";
+  import { EarlyFeature } from "@guardian/cdk/experimental";
+  ```
+
+  As the experiments are included in the `@guardian/cdk` package, compatibility is a given.
+
+  There is a risk that usage of an experimental feature goes unnoticed however, especially as IDEs typically fold import statements.
+
+- :white_check_mark: Long term maintenance
+
+  An additional directory has minimal maintenance cost.
```

Comment:
How will changes to the experimental folder be reflected in the semantic versioning? For example when you release a change that adds some experiments but does not change anything else how is that reflected in the release version?

Reply:
Releases will be made "as normal": new experiments will create a minor release, and fixes a minor. For those not making use of an experiment, Dependabot raised PRs to bump the version of `@guardian/cdk` being used should still be trusted and merged.

Reply:
Might be worth adding that to this ADR.